src/algebraic_geometry/prime_spectrum.lean

```diff
@@ -0,0 +1,202 @@
+is the set of all prime ideals of the ring that contain the set `s`.
+
+An element `f` of `R` can be thought of as a dependent function
+on the prime spectrum of `R`.
+At a point `x` (a prime ideal)
+the function (i.e., element) `f` takes values in the quotient ring `R` modulo the prime ideal `x`.
+In this manner, `zero_locus s` is exactly the subset of `prime_spectrum R`
+where all "functions" in `s` vanish simultaneously.
+-/
+def zero_locus (s : set R) : set (prime_spectrum R) :=
+{x | s ⊆ x.as_ideal}
+
+@[simp] lemma mem_zero_locus (x : prime_spectrum R) (s : set R) :
+  x ∈ zero_locus s ↔ s ⊆ x.as_ideal := iff.rfl
+
+lemma zero_locus_empty_of_one_mem (s : set R) (h : (1:R) ∈ s) :
+  zero_locus s = ∅ :=
+begin
+  rw set.eq_empty_iff_forall_not_mem,
+  intros x hx,
+  rw mem_zero_locus at hx,
+  have x_prime : x.as_ideal.is_prime := by apply_instance,
+  have eq_top : x.as_ideal = ⊤, { rw ideal.eq_top_iff_one, exact hx h },
+  apply x_prime.1 eq_top,
+end
+
+@[simp] lemma zero_locus_univ :
+  zero_locus (set.univ : set R) = ∅ :=
+zero_locus_empty_of_one_mem _ (set.mem_univ 1)
+
+@[simp] lemma zero_locus_span (s : set R) :
+  zero_locus (ideal.span s : set R) = zero_locus s :=
+begin
+  ext x,
+  simp only [mem_zero_locus],
+  split,
+  { exact set.subset.trans ideal.subset_span },
+  { intro h, rwa ← ideal.span_le at h }
+end
+
+lemma union_zero_locus_ideal (I J : ideal R) :
+  zero_locus (I : set R) ∪ zero_locus J = zero_locus (I ⊓ J : ideal R) :=
+begin
+  ext x,
+  split,
+  { rintro (h|h),
+    all_goals
+    { rw mem_zero_locus at h ⊢,
+      refine set.subset.trans _ h,
+      intros r hr, cases hr, assumption } },
+  { rintro h,
+    rw set.mem_union,
+    simp only [mem_zero_locus] at h ⊢,
+    -- TODO: The rest of this proof should be factored out.
+    rw classical.or_iff_not_imp_right,
+    intros hs r hr,
+    rw set.not_subset at hs,
+    rcases hs with ⟨s, hs1, hs2⟩,
+    apply (ideal.is_prime.mem_or_mem (by apply_instance) _).resolve_left hs2,
+    apply h,
+    rw [submodule.mem_coe, submodule.mem_inf],
+    split,
+    { exact ideal.mul_mem_left _ hr },
+    { exact ideal.mul_mem_right _ hs1 } }
+end
+
+lemma union_zero_locus (s t : set R) :
+  zero_locus s ∪ zero_locus t = zero_locus ((ideal.span s) ⊓ (ideal.span t) : ideal R) :=
+by { rw ← union_zero_locus_ideal, simp }
+
+lemma zero_locus_Union {ι : Type*} (s : ι → set R) :
+  zero_locus (⋃ i, s i) = (⋂ i, zero_locus (s i)) :=
+by { ext x, simp only [mem_zero_locus, set.mem_Inter, set.Union_subset_iff] }
+
+lemma snter_zero_locus {ι : Type*} (s : ι → set R) :
+  (⋂ i, zero_locus (s i)) = zero_locus (⋃ i, s i) :=
+(zero_locus_Union s).symm
+
+/-- The Zariski topology on the prime spectrum of a commutative ring
+is defined via the closed sets of the topology:
+they are exactly those sets that are the zero locus of a subset of the ring. -/
+instance Zariski_topology : topological_space (prime_spectrum R) :=
```

Comment:
```suggestion
instance zariski_topology : topological_space (prime_spectrum R) :=
```

Reply:
I don't know exactly what the policy is on capitalisation of names, but I think we usually don't do it.

Reply:
I don't know exactly what the policy is on capitalisation of names, but I think we usually don't do it.